### ISSUE TITLE
[Graphics] Embedded resource is ogl.mgfxo + .fx error

### ISSUE THREAD

**Issue by Devion** [craftworkgames/MonoGame.Extended#351]
Since a few days the defaultEffect in EffectResource is 'MonoGame.Extended.Graphics.Effects.Resources.DefaultEffect.ogl.mgfxo'.

This throws an immediate exception "_This MGFX effect was built for a different platform!_" when using the SharpDX monogame variant. 

Also;
VertexShaderOutputPosition VertexShaderFunctionPositionColor(VertexShaderInputPositionColor input)
{
	VertexShaderOutputPositionColor output;
	output.Position = mul(input.Position, WorldViewProjection);
	output.Color = input.Color;
	return output;
}

Throws errors due to the output returned not being of the same struct.

**Comment by craftworkgames** [comment]
I'm actually not too familiar with this part of the library but thanks for letting us know.

**Comment by LithiumToast** [comment]
@Devion

> This MGFX effect was built for a different platform!

If you want to use DirectX, you would need to use `DefaultEffect.dx11.mgfxo` instead. How to do so is not obvious, but it can be done with `EffectResource` and one of the constructors for an `Effect`. I left the constructors there for `DefaultEffect` as well. Anyways, this is one of the problems of having a portable class library with MonoGame; don't know the environment at runtime and thus can't universally decide which effect bytecode gets loaded by default. By default it loads OpenGL effect bytecode using `DefaultEffect.ogl.mgfxo` since DesktopGL is the platform used in the demos. 

> Throws errors due to the output returned not being of the same struct.

This does indeed seem to be a silly typo error [here](https://github.com/craftworkgames/MonoGame.Extended/blob/develop/Source/MonoGame.Extended.Graphics/Effects/Resources/DefaultEffect.fx#L40) on my part.

@craftworkgames Probably easier if you just fixed that typo. The effects will have to be re-built on Windows though. You can do so by running [`RebuildEffects.bat`](https://github.com/craftworkgames/MonoGame.Extended/blob/develop/Source/MonoGame.Extended.Graphics/Effects/Resources/RebuildEffects.bat).

**Comment by Devion** [comment]
Manually fixed it for myself at the moment, using the constructor and applying the DX11 version - Have to dig in deeper as tiled maps are not being rendered anymore.  (Starting to think I should switch to GL ;))

**Comment by LithiumToast** [comment]
Tiled map geometry is indexed vertices with position and texture attributes. So if you are using `DefaultEffect` make sure `VertexColorEnabled` is set to `false` and `TextureEnabled` is set to `true`. It will then use the `VertexShaderOutputPositionTexture` variation of the vertex shader and pixel shader seen in the HLSL source code.

**Comment by Devion** [comment]
No luck, going to see if openGL variant works out of the box with latest pull.

**Comment by Jjagg** [comment]
This issue can be resolved by using reflection to get the shader profile and provide the right version of an effect. https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Shader/Shader.cs#L47

**Comment by LithiumToast** [comment]
@Jjagg Cool, I'll look into this soon.

**Comment by Jjagg** [comment]
Something like the following should work.
```
// use reflection to figure out if Shader.Profile is OpenGL (0) or DirectX (1)
var mgAssembly = Assembly.GetAssembly(typeof (Game));
var shaderType = mgAssembly.GetType("Microsoft.Xna.Framework.Graphics.Shader");
var profileProperty = shaderType.GetProperty("Profile");
var value = (int) profileProperty.GetValue(null);
var extension = value == 1 ? "dx11" : "ogl";
```

**Comment by LithiumToast** [comment]
Fixed this in #369
